Bottom: d0822be85c5a453583a009ff631a9c5095ce33f3
Top:    161ed1ab1748ac66c05b2ac61493d25156f38b5e
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-01-27 22:13:39 +0000

Refresh of re-arrange-some-code

---

diff --git a/src/change.c b/src/change.c
index cb53b47..95128d0 100644
--- a/src/change.c
+++ b/src/change.c
@@ -440,21 +440,20 @@ ChangeVia2ndSize (PinTypePtr Via)
     {
       AddObjectTo2ndSizeUndoList (VIA_TYPE, Via, Via, Via);
       EraseVia (Via);
+      RestoreToPolygon (PCB->Data, VIA_TYPE, Via, Via);
       Via->DrillingHole = value;
       if (TEST_FLAG (HOLEFLAG, Via))
 	{
-	  RestoreToPolygon (PCB->Data, VIA_TYPE, Via, Via);
 	  AddObjectToSizeUndoList (VIA_TYPE, Via, Via, Via);
 	  Via->Thickness = value;
-	  ClearFromPolygon (PCB->Data, VIA_TYPE, Via, Via);
 	}
+      ClearFromPolygon (PCB->Data, VIA_TYPE, Via, Via);
       DrawVia (Via, 0);
       return (Via);
     }
   return (NULL);
 }
 
-
 /* ---------------------------------------------------------------------------
  * changes the clearance size of a via 
  * returns TRUE if changed
@@ -629,15 +628,15 @@ ChangeElement2ndSize (ElementTypePtr Element)
 	changed = true;
 	AddObjectTo2ndSizeUndoList (PIN_TYPE, Element, pin, pin);
 	ErasePin (pin);
+	RestoreToPolygon (PCB->Data, PIN_TYPE, Element, pin);
 	pin->DrillingHole = value;
-	DrawPin (pin, 0);
 	if (TEST_FLAG (HOLEFLAG, pin))
 	  {
-	    RestoreToPolygon (PCB->Data, PIN_TYPE, Element, pin);
 	    AddObjectToSizeUndoList (PIN_TYPE, Element, pin, pin);
 	    pin->Thickness = value;
-	    ClearFromPolygon (PCB->Data, PIN_TYPE, Element, pin);
 	  }
+	ClearFromPolygon (PCB->Data, PIN_TYPE, Element, pin);
+	DrawPin (pin, 0);
       }
   }
   END_LOOP;
@@ -666,15 +665,15 @@ ChangePin2ndSize (ElementTypePtr Element, PinTypePtr Pin)
     {
       AddObjectTo2ndSizeUndoList (PIN_TYPE, Element, Pin, Pin);
       ErasePin (Pin);
+      RestoreToPolygon (PCB->Data, PIN_TYPE, Element, Pin);
       Pin->DrillingHole = value;
-      DrawPin (Pin, 0);
       if (TEST_FLAG (HOLEFLAG, Pin))
 	{
-	  RestoreToPolygon (PCB->Data, PIN_TYPE, Element, Pin);
 	  AddObjectToSizeUndoList (PIN_TYPE, Element, Pin, Pin);
 	  Pin->Thickness = value;
-	  ClearFromPolygon (PCB->Data, PIN_TYPE, Element, Pin);
 	}
+      ClearFromPolygon (PCB->Data, PIN_TYPE, Element, Pin);
+      DrawPin (Pin, 0);
       return (Pin);
     }
   return (NULL);
